src/main/java/net/snowflake/ingest/streaming/internal/SnowflakeStreamingIngestClientInternal.java

```diff
@@ -162,16 +161,16 @@ public class SnowflakeStreamingIngestClientInternal<T> implements SnowflakeStrea
       SnowflakeURL accountURL,
       Properties prop,
       CloseableHttpClient httpClient,
-      boolean isIcebergMode,
       boolean isTestMode,
       RequestBuilder requestBuilder,
       Map<String, Object> parameterOverrides) {
+    this.isIcebergMode =
```

Comment:
why should isIcebergMode not be treated exactly the same as other parameters, and be handled by ParameterProvider?
Your current fix does not allow `parameterOverrides` to set isIceberg, and forces setting it via config file only.
This makes the isIceberg setting behave differently from every other setting.

Lets also not bunch together all these fixes into one PR, we were doing it to push things through but now that things are in a better state lets please go back to one PR for every logical change.

Reply:
Make sense, will split this PR. Also I think most parameters in `ParameterProvider` are not intended to be set by the customer in typical use cases. For example `account`, `role`, `authorization_type`, etc. these client level parameters can only be via config.

Reply:
Discussed offline. Keep it configureable from both props and parameterOverride.